pre_commit/languages/docker.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import unicode_literals
+
+import hashlib
+import os
+
+from pre_commit.languages import helpers
+from pre_commit.util import clean_path_on_failure
+from pre_commit.util import mkdirp
+from pre_commit.xargs import xargs
+
+
+ENVIRONMENT_DIR = 'docker'
+
+
+def md5(s):
+    m = hashlib.md5()
+    m.update(s)
+    return m.hexdigest()
+
+
+def docker_tag(repo_cmd_runner):
+    return 'pre-commit-{}'.format(
+        md5(os.path.basename(repo_cmd_runner.path()))
+    ).lower()
+
+
+def install_environment(
+        repo_cmd_runner,
+        version='default',
+        additional_dependencies=(),
+):
+    assert repo_cmd_runner.exists('Dockerfile')
+    # I don't know of anybody trying to juggle multiple docker installations
```

Comment:
I would `assert version == 'default'` with an explanation that we're not going to support multiple docker versions